### PR TITLE
Magic mirror: Fixes changing races & code improvement

### LINKED ISSUE #63064: Pride mirror race change is broken
Reporting client version: 514.1568

## Round ID:
[173666](https://scrubby.melonmesa.com/round/173666)

## Reproduction:

Pride mirror does not work for changing race anymore and has been for a while. After choosing the race it just closes the prompt and does nothing. The other options seems to still work properly.

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

All mirrors now use the same Initialize(), with a var determining which flag to check.
Improves some of the code with better var names and early returns.
Moves pride ruin's mirror to the rest of the mirrors
Removes /curse() proc, replaces it with Pride's mirror just calling parent instead, since it was the only mirror that used it.
Also fixes not being able to change races with the mirror.

## Why It's Good For The Game

Bug fix and code improvement, I hated that pride's mirror used /New() while all other mirrors used /Initialize().

Closes #63064

## Changelog

:cl:
fix: Magic mirrors will now allow you to swap species again.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/mirror.dm [hunk 1]
```diff
@@ -123,149 +121,169 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/mirror, 28)
...
+
+/obj/structure/mirror/magic/pride/attack_hand(mob/user, list/modifiers)
+	. = ..()
+	if(.)
+		return TRUE
+
+	user.visible_message(span_danger("<B>The ground splits beneath [user] as [user.p_their()] hand leaves the mirror!</B>"), \
+	span_notice("Perfect. Much better! Now <i>nobody</i> will be able to resist yo-"))
+
+	var/turf/T = get_turf(user)
```
Comment (tralezab) [inline code/game/objects/structures/mirror.dm:280]: heck make this descriptive too

File: code/game/objects/structures/mirror.dm [hunk 2]
```diff
@@ -123,149 +121,169 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/mirror, 28)
...
+	var/list/levels = SSmapping.levels_by_trait(ZTRAIT_SPACE_RUINS)
+	var/turf/dest
+	if(levels.len)
+		dest = locate(T.x, T.y, pick(levels))
+
+	T.ChangeTurf(/turf/open/chasm, flags = CHANGETURF_INHERIT_AIR)
+	var/turf/open/chasm/C = T
```
Comment (tralezab) [inline code/game/objects/structures/mirror.dm:287]: and this

**Review by tralezab — CHANGES_REQUESTED** [review]
great cleanup all around, didn't notice anything weird. also, thanks for removing the spaces on the defines

**Comment by LemonInTheDark** [comment]
@tralezab ready?

**Comment by tralezab** [comment]
Yes.

**Pull request merged**